phonetest.py

```diff
@@ -200,18 +201,93 @@ def lookup(cls, phoneid, config_file, chunk):
     @classmethod
     def match(cls, tests=None, test_name=None, phoneid=None,
               config_file=None, job_guid=None,
-              build_url=None):
+              build_url=None, branch=None, revision=None):
 
         logger.debug('PhoneTest.match(tests: %s, test_name: %s, phoneid: %s, '
                      'config_file: %s, job_guid: %s, '
-                     'build_url: %s' % (tests, test_name, phoneid,
-                                        config_file, job_guid,
-                                        build_url))
+                     'build_url: %s, branch: %s, revision: %s' % (
+                      tests, test_name, phoneid, config_file, job_guid,
+                      build_url, branch, revision))
+
         matches = []
         if not tests:
             tests = [PhoneTest.instances[key] for key in PhoneTest.instances.keys()]
 
+        # See if we need to check changeset files so we don't hit the hg
+        # server unnecessarily.
+        has_run_if_changed = False
         for test in tests:
+            if test.run_if_changed:
+                has_run_if_changed = True
+                break
+
+        def get_changeset_files(branch, revision):
+            changeset_files = set()
+            if branch in ['aurora', 'beta']:
+                branch = 'releases/mozilla-' + branch
+            elif branch not in ['mozilla-central']:
+                branch = 'integration/' + branch
+            s = requests.Session()
+            url = 'https://hg.mozilla.org/' + branch + '/json-pushes?changeset=' + revision
```

Comment:
You can get the url from the build_data['changeset'] value and use that to get the path.

Also, you can use utils.get_remote_json to retrieve this.